Scripting: Remove general cache settings

stu-elastic opened the pull request:
Removed settings:
 * `script.cache.max_size`
 * `script.cache.expire`
 * `script.max_compilations_rate`

Refs: #50152

stu-elastic committed:
Scripting: Remove general cache settings

Refs: #50152

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Scripting)

jdconrad commented on server/src/main/java/org/elasticsearch/script/ScriptCacheStats.java:
```diff
@@ -121,13 +121,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
-     * Get the general stats, null if using context cache
-     */
-    public ScriptStats getGeneralStats() {
-        return general;
```
Please also remove the member variable for this as well.

stu-elastic replied:
Removed.

stu-elastic commented on server/src/main/java/org/elasticsearch/script/ScriptService.java:
```diff
@@ -682,16 +605,10 @@ ScriptCache get(String context) {
         }
 
         ScriptStats stats() {
-            if (general != null) {
```
TODO: is this necessary?

stu-elastic replied:
Removed.

stu-elastic commented on server/src/main/java/org/elasticsearch/script/ScriptService.java:
```diff
@@ -682,16 +605,10 @@ ScriptCache get(String context) {
         ScriptStats stats() {
-            if (general != null) {
-                return general.stats();
-            }
```
May be able to remove the `ScriptStats.sum`

stu-elastic replied:
Removed

jdconrad reviewed (approved):
LGTM

stu-elastic committed:
Scripting: Remove general cache settings, update node stats

Removed settings:
 * `script.cache.max_size`
 * `script.cache.expire`
 * `script.max_compilations_rate`

Updated `_nodes/stats`:
 * Remove `script_cache`
 * Update `script` in `_node/stats` to include stats per context:
```
      "script": {
        "compilations": 1,
        "cache_evictions": 0,
        "compilation_limit_triggered": 0,
        "contexts":[
          {
            "context": "aggregation_selector",
            "compilations": 0,
            "cache_evictions": 0,
            "compilation_limit_triggered": 0
          },

```

Refs: #50152

stu-elastic committed:
Revert "Scripting: Remove general cache settings, update node stats"

This reverts commit 72f5b2b7560edc47865650eaa6768ee35d39afe7.

stu-elastic committed:
Revert "Revert "Scripting: Remove general cache settings, update node stats""

This reverts commit 1e43550c730b3b74fbbdbf8985a2a2c94c3ffe2d.
